src/plugins/kibana_utils/public/storage/hashed_item_store/hashed_item_store.ts

```diff
@@ -71,34 +71,24 @@
  */
 
 import { pull, sortBy } from 'lodash';
+import { IStorage } from '../types';
 
 interface IndexedItem {
   hash: string;
   touched?: number; // Date.now()
 }
 
-export class HashedItemStore {
+export class HashedItemStore implements IStorage<string, boolean> {
   static readonly PERSISTED_INDEX_KEY = 'kbn.hashedItemsIndex.v1';
-  private sessionStorage: Storage;
-
-  // Store indexed items in descending order by touched (oldest first, newest last). We'll use
-  // this to remove older items when we run out of storage space.
-  private indexedItems: IndexedItem[] = [];
+  private storage: Storage;
 
   /**
    * HashedItemStore uses objects called indexed items to refer to items that have been persisted
-   * in sessionStorage. An indexed item is shaped {hash, touched}. The touched date is when the item
+   * in storage. An indexed item is shaped {hash, touched}. The touched date is when the item
    * was last referenced by the browser history.
    */
-  constructor(sessionStorage: Storage) {
-    this.sessionStorage = sessionStorage;
-
-    // Potentially restore a previously persisted index. This happens when
-    // we re-open a closed tab.
-    const persistedItemIndex = this.sessionStorage.getItem(HashedItemStore.PERSISTED_INDEX_KEY);
-    if (persistedItemIndex) {
-      this.indexedItems = sortBy(JSON.parse(persistedItemIndex) || [], 'touched');
-    }
```

Comment:
Curious why did you move this out of the constructor and into the `getIndexedItems`?

Reply:
In previous pr (#51835) after moving stuff around I had to add this line in almost unrelated test file - https://github.com/elastic/kibana/pull/51835/files#diff-eba3b39d87791792aed9691f55fa23a2R21 .

Our jest config doesn't have global session/local storage mocks and that test started to pull HashedItemStoreSingleton, which tried to use sessionStorage and failed.

I decided to do that initial sorting lazily, to have less implicit dependencies like this.